### PR TITLE
add Ignore option

### PR DESCRIPTION
**Pull request by Drakirus**
See #360

### CODE REVIEW

File: autoload/leaderf.vim [hunk 1]
```diff
@@ -69,6 +69,7 @@ endfunction
 call s:InitVar('g:Lf_WindowHeight', '0.5')
 call s:InitVar('g:Lf_TabpagePosition', 2)
 call s:InitVar('g:Lf_ShowRelativePath', 1)
+call s:InitVar('g:Lf_IgnoreBufferName', 1)
```
Comment (Yggdroot) [inline autoload/leaderf.vim]: Maybe we can modify the name as `g:Lf_IgnoreCurBufferName`.

File: doc/leaderf.txt [hunk 2]
```diff
@@ -624,6 +624,12 @@ g:Lf_EmptyQuery                                   *g:Lf_EmptyQuery*
 
     Default value is 1.
 
+g:Lf_IgnoreBufferName                             *g:Lf_IgnoreBufferName*
+    This option specifies whether to remove the current buffer name from
+    the result list.
+
+    Default value is 1.
```
Comment (Yggdroot) [inline doc/leaderf.txt]: I don't this feature is enabled by default, I think it's better that the default value is 0.

File: autoload/leaderf/python/leaderf/instance.py [hunk 3]
```diff
@@ -349,6 +350,14 @@ def _actualLength(self, buffer):
         return num
 
     def setBuffer(self, content):
+        if lfEval("g:Lf_IgnoreBufferName") == '1':
+            buffer_name = os.path.normpath(lfDecode(self._cur_buffer_name))
```
Comment (Yggdroot) [inline autoload/leaderf/python/leaderf/instance.py]: On Windows, there may be a bug here. The path seperator of a normalized path is \, the path seperator in the content may be /.

File: autoload/leaderf/python/leaderf/instance.py [hunk 4]
```diff
@@ -349,6 +350,14 @@ def _actualLength(self, buffer):
...
+            if lfEval("g:Lf_ShowRelativePath") == '1':
+                buffer_name = os.path.relpath(buffer_name)
+
+            if buffer_name in content:
```
Comment (Yggdroot) [inline autoload/leaderf/python/leaderf/instance.py]: This is a time consuming operation if the length of content is very large.

**Comment by Drakirus** [comment]
Thanks for considering this feature!
As my comment on #360 says, I don't mind at all!

**Pull request closed without merging**